boards/DCM/Src/App/torquevectoring/App_TorqueVectoring.c

```diff
@@ -0,0 +1,148 @@
+#include "torquevectoring/App_TorqueVectoring.h"
+#include "torquevectoring/App_TorqueVectoringConstants.h"
+#include "torquevectoring/App_PowerLimiting.h"
+#include "torquevectoring/App_ActiveDifferential.h"
+#include "torquevectoring/App_TractionControl.h"
+#include "torquevectoring/App_Regen.h"
+#include "App_Timer.h"
+#include "App_CanRx.h"
+#include "App_CanTx.h"
+#include "App_SharedMacros.h"
+
+TimerChannel pid_timeout;
+
+PowerLimiting_Inputs       power_limiting_inputs;
+ActiveDifferential_Inputs  active_differential_inputs;
+ActiveDifferential_Outputs active_differential_outputs;
+TractionControl_Inputs     traction_control_inputs;
+TractionControl_Outputs    traction_control_outputs;
+Regen_Inputs               regen_inputs;
+Regen_Outputs              regen_outputs;
+
+PID pid_power_correction;
+// NOTE: Correction factor centered about 0.0f
+float pid_power_correction_factor = 0.0f;
+PID   pid_traction_control;
+
+float accelerator_pedal_percent;
+float wheel_speed_front_left_kph;
+float wheel_speed_front_right_kph;
+float motor_speed_left_rpm;
+float motor_speed_right_rpm;
+float battery_voltage;
+float current_consumption;
+float left_motor_temp_C;
+float right_motor_temp_C;
+float available_battery_power_kW;
+float steering_angle_deg;
+
+void App_TorqueVectoring_Setup(void)
+{
+    App_PID_Init(&pid_power_correction, &PID_POWER_CORRECTION_CONFIG);
+    App_PID_Init(&pid_traction_control, &PID_TRACTION_CONTROL_CONFIG);
+    traction_control_inputs.pid = &pid_traction_control;
+
+    App_Timer_InitTimer(&pid_timeout, PID_TIMEOUT_ms);
+}
+
+void App_TorqueVectoring_Run(void)
+{
+    // Read data from CAN
+    // NOTE: Pedal percent CAN is in range 0.0-100.0%
+    accelerator_pedal_percent   = App_CanRx_FSM_Apps_PappsMappedPedalPercentage_Get() * 0.01f;
+    wheel_speed_front_left_kph  = App_CanRx_FSM_Wheels_LeftWheelSpeed_Get();
+    wheel_speed_front_right_kph = App_CanRx_FSM_Wheels_RightWheelSpeed_Get();
+    motor_speed_left_rpm        = (float)App_CanRx_INVL_MotorPositionInfo_MotorSpeed_Get();
+    motor_speed_right_rpm       = (float)App_CanRx_INVR_MotorPositionInfo_MotorSpeed_Get();
+    battery_voltage             = App_CanRx_BMS_TractiveSystem_TsVoltage_Get();
+    current_consumption         = App_CanRx_BMS_TractiveSystem_TsCurrent_Get();
+    left_motor_temp_C           = App_CanRx_INVL_Temperatures3_MotorTemperature_Get();
+    right_motor_temp_C          = App_CanRx_INVR_Temperatures3_MotorTemperature_Get();
+    // TODO(akoen): Available power will soon be replaced by current + voltage messages
+    available_battery_power_kW = App_CanRx_BMS_AvailablePower_AvailablePower_Get();
+    steering_angle_deg = App_CanRx_FSM_Steering_SteeringAngle_Get();
+
+    if (accelerator_pedal_percent > 0.01f)
```

Comment:
Maybe something a bit higher than this is safer, like 10%
Also note that PappsMappedPedalPercentage is sent in percent so 0-100, so this value above here is 0.01%

Reply:
Set it to 10%, although note that the percentage is scaled 0-1 above